modules/nextflow/src/main/groovy/nextflow/extension/PublishOp.groovy

```diff
@@ -98,26 +105,65 @@ class PublishOp {
             publisher.apply(files, sourceDir)
         }
 
-        // create record for index file
+        // append record to index file
         if( indexOpts ) {
             final record = indexOpts.mapper != null ? indexOpts.mapper.call(value) : value
-            final normalized = normalizePaths(record)
+            final normalized = normalizePaths(record, targetDirOrClosure)
```

Comment:
Ok, it has not been introduced in this PR, this always though this is not scalable with a large number of files. It should be taken in consideration an stream like approach before it's released as stable e.g. [here](https://github.com/nextflow-io/nextflow/blob/fix-workflow-outputs/modules/nextflow/src/main/groovy/nextflow/cache/CacheDB.groovy#L50-L50)

Reply:
Good point, I will address this in a follow-up PR